packages/fsapp/src/types/index.ts

```diff
@@ -83,3 +84,10 @@ export interface NavigatorEvent {
   id: string;
   type: string;
 }
+
+export interface Location<T = any> {
+  pathname: string;
+  search: string;
+  hash: string;
+  state: T;
```

Comment:
Some of these should probably be optional. If you take the values from window.location, the strings should be defined, but state wouldn't be.

Reply:
I agree this is wonky, but unfortunately this is a copy of @types/history which is used by react-native-router: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/history/index.d.ts#L28

There were some dependency concerns with importing @types/history into flagship, so I decided to make my own interface since I don't think the history API is going to change anytime soon. :)

We're just going to have to pass empty strings and objects to satisfy the types in the router.

Reply:
Makes sense.